Change Dockerfile to a dual-staged build/run approach

rogbas opened the pull request:
Pushing this Dockerfile to integrate with quay.io pipeline, and to test again ci-operator integration

jewzaam commented on build/Dockerfile:
```diff
@@ -1,7 +1,13 @@
+ENV OPERATOR_PATH=/go/src/github.com/openshift/dedicated-admin-operator
+COPY . ${OPERATOR_PATH}
+WORKDIR ${OPERATOR_PATH}
+RUN make build
```
How about `RUN yum update -y && yum clean all && make build` ?

jewzaam replied:
Guess would exclude go in the update.  Unless we don't care about the version of go.

rogbas replied:
done, excluding the golang package from the yum update

jewzaam commented on build/Dockerfile:
```diff
@@ -1,7 +1,13 @@
-USER nobody
-
-ADD build/_output/bin/dedicated-admin-operator /usr/local/bin/dedicated-admin-operator
+FROM alpine:latest  
```
Not pinning to a version is a risk in my mind, that we don't know what we're going to get.  Prefer go back to a specific tag, you had `alpine:3.8` before.  While on that, any reason why we're staying on `alpine` and not looking for a RH, openshift, or origin base image?

lisa replied:
Agreed, we want to pin to a specific version, whether it's alpine or something else.

rogbas replied:
Ok, I'll lock on 3.8. It's alpine mostly cause the template generated by operator-sdk comes with that.

Another option would be having `FROM scratch`, thoughts?

lisa replied:
@rogbas about `scratch`, be sure to test it because it's possible that you don't have CA cert bundle, or `/etc/passwd` or other bits we take for granted which are relied upon by syscalls or libraries.

rogbas replied:
That's true @lisa, just locked the alpine version for now.

rogbas commented:
Changes made:
* Lock apline version to 3.8
* Added yaml update/clean
* Added user/UID config (based on latest Dockerfile generated by operator-sdk)

rogbas commented:
Build notes added to the README file. 
Also added LABELS to the Dockerfile to tag the img itself
